src/utils/tcti.rs

```diff
@@ -0,0 +1,450 @@
+                .or(Err(Error::WrapperError(WrapperErrorKind::InvalidParam)))
+        }
+    }
+}
+
+impl FromStr for Tcti {
+    type Err = Error;
+
+    fn from_str(config_str: &str) -> Result<Self, Self::Err> {
+        let device_pattern = Regex::new(r"^device(:(.*))?$").unwrap(); //should not fail
+        if let Some(captures) = device_pattern.captures(config_str) {
+            return Ok(Tcti::Device(DeviceConfig::from_str(
+                captures.get(2).map_or("", |m| m.as_str()),
+            )?));
+        }
+
+        let mssim_pattern = Regex::new(r"^mssim(:(.*))?$").unwrap(); //should not fail
+        if let Some(captures) = mssim_pattern.captures(config_str) {
+            return Ok(Tcti::Mssim(MssimConfig::from_str(
+                captures.get(2).map_or("", |m| m.as_str()),
+            )?));
+        }
+
+        let tabrmd_pattern = Regex::new(r"^tabrmd(:(.*))?$").unwrap(); //should not fail
+        if let Some(captures) = tabrmd_pattern.captures(config_str) {
+            return Ok(Tcti::Tabrmd(TabrmdConfig::from_str(
+                captures.get(2).map_or("", |m| m.as_str()),
+            )?));
+        }
+
+        Err(Error::WrapperError(WrapperErrorKind::InvalidParam))
+    }
+}
+
+#[test]
+fn validate_from_str_tcti() {
+    let tcti = Tcti::from_str("mssim:port=1234,host=168.0.0.1").unwrap();
+    assert_eq!(
+        tcti,
+        Tcti::Mssim(MssimConfig {
+            port: 1234,
+            host: ServerAddress::Ip(IpAddr::V4(std::net::Ipv4Addr::new(168, 0, 0, 1)))
+        })
+    );
+
+    let tcti = Tcti::from_str("mssim").unwrap();
+    assert_eq!(
+        tcti,
+        Tcti::Mssim(MssimConfig {
+            port: DEFAULT_SERVER_PORT,
+            host: Default::default()
+        })
+    );
+
+    let tcti = Tcti::from_str("device:/try/this/path").unwrap();
+    assert_eq!(
+        tcti,
+        Tcti::Device(DeviceConfig {
+            path: PathBuf::from("/try/this/path"),
+        })
+    );
+
+    let tcti = Tcti::from_str("device").unwrap();
+    assert_eq!(tcti, Tcti::Device(Default::default()));
+
+    let tcti = Tcti::from_str("tabrmd:bus_name=some.bus.Name2,bus_type=session").unwrap();
+    assert_eq!(
+        tcti,
+        Tcti::Tabrmd(TabrmdConfig {
+            bus_name: String::from("some.bus.Name2"),
+            bus_type: BusType::Session
+        })
+    );
+
+    let tcti = Tcti::from_str("tabrmd").unwrap();
+    assert_eq!(tcti, Tcti::Tabrmd(Default::default()));
+}
+
+/// Configuration for a Device TCTI context
+///
+/// The default configuration uses the library default of
+/// `/dev/tpm0`.
+#[derive(Clone, Debug, PartialEq)]
+pub struct DeviceConfig {
+    /// Path to the device node to connect to
+    ///
+    /// If set to `None`, the default location is used
+    path: PathBuf,
+}
+
+impl Default for DeviceConfig {
+    fn default() -> Self {
+        DeviceConfig {
+            path: PathBuf::from("/dev/tpm0"),
+        }
+    }
+}
```

Comment:
🆗

Reply:
I'm legit afraid that would need to be changed to be OS-dependent

Reply:
I share your concern but I think it is ok for now.